src/diffusers/pipelines/pag/pag_utils.py

```diff
@@ -0,0 +1,260 @@
+# Copyright 2024 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+
+from ...models.attention_processor import (
+    PAGCFGIdentitySelfAttnProcessor2_0,
+    PAGIdentitySelfAttnProcessor2_0,
+)
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+
+class PAGMixin:
```

Comment:
Why is this not under `loaders`?

Reply:
Gentle ping @yiyixuxu in case this got missed.

Reply:
so `PAGMixin` is a little bit different:
1. it is more of a util (so it is not a loader, it does not load weights, it is similarly to the `StableDiffusionMixin` here https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/pipeline_utils.py#L1883)
2. I think it will only used by PAG pipelines under this PAG folder so I think we can keep it in the PAG folder. no?

Reply:
Yessss! Thanks for explaining